packages/material-ui-lab/src/Masonry/Masonry.js

```diff
@@ -81,48 +106,84 @@ const Masonry = React.forwardRef(function Masonry(inProps, ref) {
   });
 
   const masonryRef = React.useRef();
+  const [maxColumnHeight, setMaxColumnHeight] = React.useState();
+  const [maxNumOfRows, setMaxNumOfRows] = React.useState();
   const { children, className, component = 'div', columns = 4, spacing = 1, ...other } = props;
-  const ownerState = { ...props, spacing, columns };
+  const ownerState = { ...props, spacing, columns, maxColumnHeight, maxNumOfRows };
   const classes = useUtilityClasses(ownerState);
+  const lineBreakStyle = {
+    flexBasis: '100%',
+    width: 0,
+    margin: 0,
+    padding: 0,
+    content: '',
+  };
 
-  const contextValue = React.useMemo(() => ({ spacing }), [spacing]);
-  let didWarn = false;
   React.useEffect(() => {
-    // scroller always appears when masonry's height goes beyond 2,000px on Chrome
-    const handleScroll = () => {
-      if (masonryRef.current.clientHeight === 1998 && !didWarn) {
-        console.warn(
-          [
-            'Material-UI: The Masonry can have the maximum height of 2,000px on Chrome browser.',
-            'Items that go beyond this height fail to be rendered on Chrome browser.',
-            'You can find more in this open issue: https://github.com/mui-org/material-ui/issues/27934',
-          ].join('\n'),
-        );
-
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-        didWarn = true;
+    const handleResize = () => {
+      const columnHeights = [];
+      const numOfRows = [];
+      let skip = false;
+      Array.from(masonryRef.current.children).forEach((child) => {
```

Comment:
If `masonryRef.current.children` is [Element.children](https://developer.mozilla.org/en-US/docs/Web/API/Element/children) then you don't need to convert it into an array since `Element.children` inherits from `NodeList` which implements `forEach` as well.


```suggestion
     masonryRef.current.children.forEach((child) => {
```

Reply:
Element.children is not an array. It is an object called an HTMLCollection. These do not have an array’s methods (though they do have the length property).

Reply:
Or just tell me that it doesn't inherit from `NodeList`. Let's use `nodeList` and check the `nodeType` for elements to avoid creating arrays.

Reply:
Ah, I replaced `children` with `childNodes`, and `forEach()` works. Thanks!